xilem_core/src/sequence.rs

```diff
@@ -785,3 +786,112 @@ impl_view_tuple!(M0, Seq0, 0; M1, Seq1, 1; M2, Seq2, 2; M3, Seq3, 3; M4, Seq4, 4
 impl_view_tuple!(M0, Seq0, 0; M1, Seq1, 1; M2, Seq2, 2; M3, Seq3, 3; M4, Seq4, 4; M5, Seq5, 5; M6, Seq6, 6; M7, Seq7, 7; M8, Seq8, 8; M9, Seq9, 9; M10, Seq10, 10; M11, Seq11, 11; M12, Seq12, 12; M13, Seq13, 13);
 impl_view_tuple!(M0, Seq0, 0; M1, Seq1, 1; M2, Seq2, 2; M3, Seq3, 3; M4, Seq4, 4; M5, Seq5, 5; M6, Seq6, 6; M7, Seq7, 7; M8, Seq8, 8; M9, Seq9, 9; M10, Seq10, 10; M11, Seq11, 11; M12, Seq12, 12; M13, Seq13, 13; M14, Seq14, 14);
 impl_view_tuple!(M0, Seq0, 0; M1, Seq1, 1; M2, Seq2, 2; M3, Seq3, 3; M4, Seq4, 4; M5, Seq5, 5; M6, Seq6, 6; M7, Seq7, 7; M8, Seq8, 8; M9, Seq9, 9; M10, Seq10, 10; M11, Seq11, 11; M12, Seq12, 12; M13, Seq13, 13; M14, Seq14, 14; M15, Seq15, 15);
+
+/// A stub `ElementSplice` implementation for `NoElement`.
+///
+/// It is technically possible for someone to create an implementation of `ViewSequence`
+/// which uses a `NoElement` `ElementSplice`. But we don't think that sequence could be meaningful.
+pub(crate) struct NoElements;
+
+impl ElementSplice<NoElement> for NoElements {
+    fn with_scratch<R>(&mut self, f: impl FnOnce(&mut AppendVec<NoElement>) -> R) -> R {
+        let mut append_vec = AppendVec::default();
+        f(&mut append_vec)
+    }
+
+    fn insert(&mut self, _: NoElement) {}
+
+    fn mutate<R>(&mut self, f: impl FnOnce(<NoElement as crate::ViewElement>::Mut<'_>) -> R) -> R {
+        f(())
+    }
+
+    fn skip(&mut self, _: usize) {}
+
+    fn delete<R>(&mut self, f: impl FnOnce(<NoElement as crate::ViewElement>::Mut<'_>) -> R) -> R {
+        f(())
+    }
+}
+
+/// The [`ViewSequence`] for [`without_elements`], see its documentation for more context.
+pub struct WithoutElements<Seq, State, Action, Context, Message> {
+    seq: Seq,
+    phantom: PhantomData<fn() -> (State, Action, Context, Message)>,
+}
+
+/// A [`ViewSequence`] that doesn't contain any elements, it can be used everywhere where a view sequence (with or without elements) is expected.
```

Comment:
This documentation is very oblique; I know why it exists, but to someone stumbling upon this, it would likely be extremely mysterious.

Perhaps we should say something like:
> An adapter which turns a [`ViewSequence`]  containing any number of views with side effects, into a `ViewSequence` with any element type.
>
> This can be used to embed side-effects naturally into the flow of your program.
> This can be used as an alternative to [`fork`] , which avoids adding extra nesting at the cost of only being usable in places where there is already an existing sequence.